### ISSUE TITLE
[BUG] Disk shows replicas that don't exist. Node shows none.

### ISSUE THREAD

**Issue by LilDrunkenSmurf** [longhorn/longhorn#5579]
## Describe the bug (🐛 if you encounter this issue)

Node shows 0 replicas, but is over-allocated (was 100% allocated with 0% used). Disk shows 104 replicas, but has an empty list.

## To Reproduce

![image](https://user-images.githubusercontent.com/46251616/225639572-c4239883-7496-4291-91ba-3e1d3d1d4efa.png)
![image](https://user-images.githubusercontent.com/46251616/225639615-421a907b-4ef9-4bab-b9e9-4b7ed6cda845.png)

## Expected behavior
Node shows total number of replicas on all disks. Replicas are properly listed.

## Log or Support bundle
[supportbundle_a33949f1-e247-4a26-a6cd-3dcac7397190_2023-03-16T13-46-44Z.zip](https://github.com/longhorn/longhorn/files/10991490/supportbundle_a33949f1-e247-4a26-a6cd-3dcac7397190_2023-03-16T13-46-44Z.zip)

## Environment

 - Longhorn version: v1.41
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): Helm
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: k3s
   - Number of management node in the cluster: 1
   - Number of worker node in the cluster: 3
 - Node config
   - OS type and version: Rasbian 64bit (Debian 11)
   - CPU per node: 4
   - Memory per node: 4GB
   - Disk type(e.g. SSD/NVMe): USB 3.0 Flash Drive
   - Network bandwidth between the nodes: Gigabit
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): Baremetal
 - Number of Longhorn volumes in the cluster: 11

## Additional context
One of my nodes randomly filled up, and won't clean itself up. I noticed then when I was having constant rebuild failures on volumes. I was trying to remove and re-add the disk since the directory was empty, but the disk in the UI shows 104 replicas (but no listed ones), and was at 100% allocated until I changed it to no longer overprovision, and is now at 187% allocation. Eviction doesn't do anything.

**Comment by LilDrunkenSmurf** [comment]
I resolved this by unmounting the disk from the host, removing it from longhorn, and re-adding it. The space was available again, but I'm having a new issue where all the volumes on this host are `stopped` no matter what I do. I'm not sure if I should file a new big for that or not.

**Comment by derekbit** [comment]
I think this is because you unmouned the disk from the host, removed it from longhorn, and re-added it before detaching it, right?
Can the volumes be reattach the volumes?

**Comment by LilDrunkenSmurf** [comment]
Sorry, what do you mean? The "new" disk seems healthy and fine, and pods running on this node can attach to volumes, but replicas are all showing as stopped.

**Comment by LilDrunkenSmurf** [comment]
I'll open a new bug with a new bundle

**Comment by deosrc** [comment]
I also encountered this I believe as a result of longhorn trying to replica volumes while my cluster was unstable. I've managed to get to the same point of the replicas being gone but new replicas always being stopped. Here's the method I used:

1. Evict all working volumes and disable scheduling. I did this at both node and disk level.
2. Navigate to the directory for the disk and delete all files. This includes the config file and the replicas. Longhorn UI should update automatically but the values won't make sense.
3. Edit the node to delete the disk. This should now be possible as longhorn detects no replicas on the disk.
4. Recreate the disk. If you get an error about the node syncing, just wait a few seconds and try again.
5. Re-enable the node.

I'm now looking at #5762 to try resolve the stopped replicas issue.
